numpy/lib/tests/test_function_base.py

```diff
@@ -902,9 +902,9 @@ def test_specific_axes(self):
         # test maximal number of varargs
         assert_raises(TypeError, gradient, x, 1, 2, axis=1)
 
-        assert_raises(ValueError, gradient, x, axis=3)
-        assert_raises(ValueError, gradient, x, axis=-3)
-        assert_raises(TypeError, gradient, x, axis=[1,])
+        assert_raises(np.AxisError, gradient, x, axis=3)
+        assert_raises(np.AxisError, gradient, x, axis=-3)
+        # assert_raises(TypeError, gradient, x, axis=[1,])
```

Comment:
This used to error, and ideally it would still error for consistency with `ufunc.reduce`. But all the functions using `_ureduce` do not error in this situation, so we've already failed at consistency.

Perhaps we should add a deprecation warning (in a future commit) about passing lists of axes?